SEO Optimizations extended

ananyo2012 opened the pull request:
@sonalkr132 I added some more SEO optimizations.Please review them.The commits are self explanatory about the changes.
Here is the seo site checkup of the app i deployed.It is updated with the changes till date along with the changes in this PR.
http://seositecheckup.com/checkup/result/f399ef297f980d4c5d3ab3a7c2aebea3?url=https://desolate-bastion-9663.herokuapp.com
3 more tests will pass namely the Media print css, Media Query Responsive and URL seo friendly test with the ui changes made in the deployment of @ghoshbishakh.
One more thing that i didn't change the sign_in paths of devise.You can see there are two urls which are said to be underscored so not seo friendly.Do i need to change them?

sonalkr132 commented on app/views/sitemap/index.xml.haml:
```diff
@@ -7,12 +12,27 @@
+  %url
+    %loc #{new_user_confirmation_url}/
+    %changefreq monthly
+    %priority 0.5
```
Are you sure we need new_user_\* paths in sitemap?

ananyo2012 replied:
well sitemaps need to include urls that contain most of the content.i am not sure about it either.then i may remove the password and confirmation url.but i think keeping the sign in  page is better.

sonalkr132 commented on public/robots.txt:
```diff
@@ -2,6 +2,8 @@
+ Disallow: /blockeds
+ Disallow: /activities
 
  Sitemap: http://www.catforum.mntc.in/sitemap.xml
```
Why remove flags?

ananyo2012 replied:
I authenticated the flags controller in the last PR so we don't need to include it in robots.txt

sonalkr132 replied:
isn't blocked, activities and admins covered with authorization as well?

ananyo2012 replied:
no blockeds, activities do not have authentication.by admins i mean the admins sign in page here.i have disallowed that too,i.e. we don't want even the admins sign in page to be crawled.

sonalkr132 replied:
> isn't blocked, activities and admins covered with **authorization** as well?

I am sorry, I mean **authentication**. And I am pretty sure you need to authenticated as admin user to access those pages.

ananyo2012 replied:
@sonalkr132 so will i add the authentication to those pages and remove the urls from robots.txt?
there is no authentication set in blockeds and activities controller.

sonalkr132 replied:
oh there isn't. We need to fix it. Only admin user should be able to see blocked users and only proper user should be able to see his activity, not everyone.

ananyo2012 replied:
ok then can i do it in this PR? yeah there isn't.but anyways they won't be able to see since activities need a current user without which it will generate an error and bockeds url is also generating some error.actually blockeds is not serving any business.all block activities is controlled by flags controller.

sonalkr132 replied:
If you are sure that blocked controller is not doing anything then you delete it. Correct process would be roll back migration file of blocked and then `rails d scaffold Blocked`

sonalkr132 replied:
You can push auth related to activity to this PR.

ananyo2012 commented:
@sonalkr132 Please review the commits.
I removed the urls from new_user_password and confirmation urls from sitemap.
Also removed activities and blocked urls from robots.txt.
For time being i also added authentication to blockeds controller as i am not confident enough to remove it now.Presently blocking is done using devise.So i have to read the doc once.

sonalkr132 commented on app/controllers/activities_controller.rb:
```diff
@@ -1,6 +1,8 @@
+    sleep 0.7
     @activities = PublicActivity::Activity.order("created_at desc").where(recipient_id: current_user).where("owner_id not in (?)", current_user).all
   end
 end
```
We don't only have to authenticate user but authorize them too.

ananyo2012 replied:
@sonalkr132 I can't find any way to define ability for index action of activities.Searched a lot but couldn't find any related answers.
I put this line in ability.rb in `permitted(user)` method  `can [:index], PublicActivity::Activity` 
and used `authorize! :index, @activities` for authorization in activities_controller.
Can you point me where i am going wrong?Its not giving any error.guest user can't access activities.But logged in user is also unable to access activities.

sonalkr132 replied:
Make sure that `PublicActivity::Activity` is not nil in your ability model. A simple `puts #{PublicActivity::Activity}` will be enough.
If it is then you will need to include public activity model in ability class. Go through public_acitivity  [doc](https://github.com/chaps-io/public_activity)

ananyo2012 replied:
@sonalkr132 tried it still no luck.Added this line to ability.rb  `include PublicActivity::Model` But it is giving an error undefined method has_many.only using `include PublicActivity` gives same problem as before.doing a puts statement outputs as it is.

sonalkr132 replied:
Leave it. It is not that big a deal. I can't think of situation where some user will be see other's activity, as you had pointed out that we were using current_user to find the activity.
